components/playlist/common/mojom/playlist.mojom

```diff
@@ -124,6 +125,15 @@ interface PlaylistService {
   AddObserver(pending_remote<PlaylistServiceObserver> observer);
 };
 
+// WebUI will delegate some UIs to native UI via this interface. While
+// PlaylistService is shared among browsers with the same profile, a NativeUI
+// instance will be bound to each web ui.
+interface PlaylistNativeUI {
+  ShowCreatePlaylistUI();
+
+  ShowRemovePlaylistUI(string playlist_id);
```

Comment:
This is precisely what we do not want. This is going to lead to inconsistencies and bugs between implementations on different platforms. It shouldn't matter whether the methods are called by a webui or native ui in the mojo interface and the UI should not have to make additional calls to get more information (like additional information based on the playlist id.

Reply:
I guess there are really two parts to this problem here. The first is that we shouldn't need code that is specific to the webui opening a native UI here. The other problem is below from previous PRs with things like `OnEvent` that require the Ui to know what to do in response to changes (vs telling the UI what to do)

Reply:
Thanks, opened a follow-up issue for this: https://github.com/brave/brave-browser/issues/33863